aip/0160.md

```diff
@@ -0,0 +1,166 @@
+---
+aip:
+  id: 160
+  state: reviewing
+  created: 2020-02-24
+permalink: /160
+redirect_from:
+  - /0160
+---
+
+# Filtering
+
+Often, when listing resources (using a list method as defined in [AIP-132][] or
+something reasonably similar), it is desirable to filter over the collection
+and only return results that the user is interested in.
+
+It is tempting to define a structure to handle the precise filtering needs for
+each API. However, filtering requirements evolve frequently, and therefore it
+is prudent to use a string field with a structured syntax accessible to a
+non-technical audience. This allows updates to be able to be made
+transparently, without waiting for UI or client updates.
+
+**Note:** Because list filters are intended for a potentially non-technical
+audience, they sometimes borrow from patterns of colloquial speech rather than
+common patterns found in code.
+
+## Guidance
+
+APIs **may** provide filtering to users on `List` methods. If they choose to do
+so, they **must** follow the common specification for filters discussed here.
+The syntax is formally defined in the [EBNF grammar][].
+
+**Note:** Google provides implementations for the filtering language in both
+[C++][cel-cpp] and [Go][cel-go] as a discrete component of [CEL][]. The use of
+these implementations is encouraged.
+
+### Expressions
+
+Expressions are literal values in the filtering syntax ("42", "Victor Hugo").
```

Comment:
This paragraph is horrible, and I spent an hour on it. Hoping to rewrite it tomorrow morning with a clear head, but I welcome ideas.

Reply:
I would just call this section literals. While the syntax discusses 'expressions' and a literal could be an expression, I wouldn't phrase this as 'expressions are literals'.

A bare literal is treated a global restriction which can be thought of like a fuzzy match against the literal anywhere it may appear in an object.

Then I would mention that you can use sequences of literals as an implicit AND between fuzzy matchers, or that the literal may appear on the right-hand side of an operator.